### ISSUE TITLE
Add a completed project of the "Animating/Controlling thousands of fish with MultiMeshInstance" tutorial for the readers to download.

### ISSUE THREAD

**Issue by Waranoi** [godotengine/godot-docs#3890]
**Your Godot version:**
3.2.1 stable

**Issue description:**
There are multiple places where it's hard to follow the tutorial, adding a link to a completed project for the tutorial like there is for the FPS tutorial would let the readers find the solution to any issues that arise by themselves.

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/stable/tutorials/3d/vertex_animation/animating_thousands_of_fish.html
https://docs.godotengine.org/en/stable/tutorials/3d/vertex_animation/controlling_thousands_of_fish.html

**Comment by Calinou** [comment]
This demo would probably be better added to the [godot-demo-projects](https://github.com/godotengine/godot-demo-projects) repository.

**Comment by clayjohn** [comment]
Could you mention which sections are hard to follow?

**Comment by Waranoi** [comment]
> Could you mention which sections are hard to follow?

1. "We will start with a single fish. Load your fish model into a MeshInstance and add a new ShaderMaterial."
![image](https://user-images.githubusercontent.com/31479779/89913140-a7626200-dbf3-11ea-823d-bff5811a6acb.png)
It's not clear where to add your new ShaderMaterial. If you add it in Material Override then the color of the whole surface gets overridden and the color of the fish becomes all wrong. I have yet to find a way to implement a vertex shader while preserving the color scheme of the fish when adding a new ShaderMaterial to Material Override. (I've read through the "Getting Started - Step by step", "3D" and "Shading" tutorials and asked for help in a Godot discord)<br>
If you add it in the Material section of the MeshInstance then you need to add three new ShaderMaterials, (the wording of the tutorial implies that you only add one ShaderMaterial) each sharing a singular shader with a color uniform so that your vertex displacements affects all surfaces and so that you can set the colors to match the original fish object. There are issues with this approach that I'll mention further down.

2. `float body = (VERTEX.z + 1.0) / 2.0; //for a fish centered at (0, 0) with a length of 2`
![image](https://user-images.githubusercontent.com/31479779/89915075-307a9880-dbf6-11ea-9a5f-b836aa226477.png)
The fish seems to be 7-8 units long out of the box, there's no mention of the mesh being scaled down in the editor. It makes the reader worried that the tutorial or the model might not be up-to-date or that the reader did a mistake somewhere or is lacking key knowledge.

3. "Putting the four motions together gives us the final animation."
The animation of my fish did not match the one shown in the tutorial. When the default values for the uniforms used in the code are left out then you can't use the same values to see if the results match and verify that you implemented the code correctly. It would also be nice to have a small section with the finished code at the end to compare with your own implementation, but having a completed project of the tutorial that you can download would serve the same purpose.

4. "The second is to loop over all the instances and set their transforms in code. Below, we use GDScript to loop over all the instances and set their transform to a random position."
It does not specify where to write the code, I assumed that you were meant to put it in the `_ready()` function.

5. "Running this script will place the fish in random positions in a box around the position of the MultiMeshInstance."
![image](https://user-images.githubusercontent.com/31479779/89916517-fb6f4580-dbf7-11ea-981f-50168ceb2201.png)
There's no Material section for the MultiMeshInstance node. (This is the issue I was talking about earlier about adding my ShaderMaterials to the Material section of MeshInstance) So I had to add my ShaderMaterials to the Surface sections of the mesh in the MultiMeshInstance node but when I run the scene my fish aren't animating. My guess is that because I'm adding the ShaderMaterials directly on my mesh's surfaces, the changes are not saved when I run the game. (I'm thinking about that big "Changes may be lost!" warning when I open the surfaces default SpatialMaterial.)

This is as far as I got, I haven't read the tutorial past this point since my fish are currently not animating correctly when I run the scene. I could set the transforms of the fishes purely through the editor as they mention in the tutorial, but then the position of the fishes won't be random every time you run the scene, compared to if you were to use the code the tutorial presented and place it in the `_ready()` function of your script.

**Comment by Atem1995** [comment]
I got the same problem Described above.Please Help!!!

**Comment by Roemer** [comment]
Did anyone ever successfully completed this manual? I am hanging on the same points as @Waranoi.  Basically at the very first step: `and add a new ShaderMaterial`
I would love to complete this tutorial and update the manual accordingly but I just cannot get it to work without a LOT of changes, which seems strange to me.

**Comment by kyle-wendling** [comment]
Still incomplete sadly. It's a just a bit strange to think that someone spent the time to put together this tutorial, which seems helpful, and didn't have the theory of mind to realize how incomplete it is for someone trying to follow, and didn't bother to include a example projects, very strange?

**Comment by solodeveloping** [comment]
> Could you mention which sections are hard to follow?

Hi, thank you for the tutorial, I just watched a tutorial about the Abzu game, and then the Abzu GDC, and it was nice to see the same was possible in Godot.

I'd like to say your work is appreciated before getting to the "rude" part.

-> It stopped working like in the tutorial at the 3rd step for the vertexes, and in the first step for the colors.

I see you're the one that made the tutorial in 2019, it's been 7 years since then, and it's hard to believe that someone has managed to make it work without a lot of adjustements.

It's understable that you're busy, and might not even see this, but, maybe you could just push the project you made quickly and let someone else adjust the tutorial then? (and I checked your profile, I could not find it)

Is the part about not being able to access the initial material colors normal? I tried many things, and I do understand that @clayjohn private versions uses a custom model, but it seem odd there are never any ways to not modify the color / keep it intact, you can't use overlay because of the vertex displacement, you can't just live "fragment" commented etc

Hopefully, this will get an update 😊

Thank you in advance 😊

**Comment by solodeveloping** [comment]
For people wondering about the colors: you'll have to make your own mesh / update Quaternius model, see https://github.com/godotengine/godot-docs/issues/6428
